.github/workflows/lint-nix.yml

```diff
@@ -0,0 +1,63 @@
+# This yml handles linting nix configs for each PR
+### Workflow
+# 1. Developer writes nix configs
+# 2. Developer pushes code up to fork and opens new PR
+# 3. This job kicks off, verifying the new configuration they added
+# 4. If the job suceeds, the PR is allowed to be merged. If not, the PR is unable to be merged
+
+name: Lint Nix Configs
+on: pull_request
+
+jobs:
+  linter:
+    name: Run nixpkgs-fmt
+    runs-on: ubuntu-18.04
+
+    # nixos/nix docker image v2.3 SHA
+    container:
+      image: nixos/nix@sha256:af330838e838cedea2355e7ca267280fc9dd68615888f4e20972ec51beb101d8
+
+    steps:
+      - name: Pulls Repository
+        uses: actions/checkout@v2
+      
+        # Utilizing an external GitHub Action to obtain all changed files
+        # https://github.com/jitterbit/get-changed-files
+        # Utilizing SHA for v1
+      - name: Get All Changed Files
+        id: filediff
+        uses: jitterbit/get-changed-files@b17fbb
+        with:
+          format: space-delimited
+      
+        # Parses through all of the changed files using nixpkgs-fmt, outputting the diff
+        # between the modded one and original if one of the new nix configs are not properly formatted
+        # NOTE: When using `diff` in GitHub Actions, it causes script to forcibly exit, so workaround is to
+        # unset e so that the script will "ignore" whatever return value is leaking in the diff command
+      - name: Verify Nix Configs
+        id: format
+        run: |
+          cd $GITHUB_WORKSPACE
+          has_failed="false"
+          for changed_file_path in ${{ steps.filediff.outputs.all }}; do
+            full_filename=$(basename $changed_file_path)
+            file_extension="${full_filename##*.}"
+            fmt_file_path="${changed_file_path%.*}_fmt.nix"
+            if [[ "$file_extension" == "nix" ]] && [[ -f "$changed_file_path" ]]; then
+              nix-shell --run "cat $changed_file_path | nixpkgs-fmt > $fmt_file_path" 
+
+              set +e
+              differences=$(diff $changed_file_path $fmt_file_path)
+              set -e
+
+              if [ ! -z "$differences" ]; then
+                printf "\n$differences\n"
```

Comment:
The output of the diff looks good for the other test results you included, thanks for implement it like this

Reply:
No problem! I wanted to make sure that the output looks similar to other tools that did that, like `terraform fmt`.